Sources/LanguageServerProtocol/WorkspaceSettings.swift

```diff
@@ -63,6 +67,15 @@ public struct ClangWorkspaceSettings: Codable, Hashable {
     self.compilationDatabasePath = compilationDatabasePath
     self.compilationDatabaseChanges = compilationDatabaseChanges
   }
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    compilationDatabasePath = try? container.decode(String.self, forKey: .compilationDatabasePath)
+    compilationDatabaseChanges = try? container.decode([String: ClangCompileCommand].self, forKey: .compilationDatabaseChanges)
+    guard (compilationDatabasePath == nil) != (compilationDatabaseChanges == nil) else {
+      throw MessageDecodingError.invalidRequest("ClangWorkspaceSettings can have one and only one valid property")
+    }
+  }
```

Comment:
Why is this necessary?

Reply:
I don't have full context of the clangd protocol usage here, but from the comment above:
```Clangd will accept *either* a path to a compilation database on disk, or the contents of a compilation database to be managed in-memory, but they cannot be mixed.```

Otherwise any null setting change sent from client will be interpreted as `ClangWorkspaceSettings`, this essentially address the TODO I deleted